File: mobile/src/services/comment.ts
import { http } from './request';
import type { Comment } from '../types';

export interface CommentDTO {
  articleId: number;
  parentId?: number;
  nickname: string;
  email?: string;
  content: string;
}

/** 某文章评论列表 GET /api/comments?articleId=x（仅已审核） */
export function getComments(articleId: number | string) {
  return http.get<Comment[]>('/comments', { articleId });
}

/** 提交评论 POST /api/comments（提交后待审核） */
export function submitComment(data: CommentDTO) {
  return http.post<void>('/comments', data);
}

File: backend/src/main/java/com/blog/entity/Comment.java
package com.blog.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 评论实体
 */
@Data
@TableName("comment")
public class Comment {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long articleId;

    /** 父评论 ID，0 为顶级 */
    private Long parentId;

    private String nickname;

    private String email;

    private String avatar;

    private String content;

    private String ip;

    /** 0 待审核 / 1 通过 / 2 拒绝 */
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}
